### ISSUE TITLE
17.06.0-ce-rc5 (and earlier 17.06 RC builds) breaks rancher-agent

### ISSUE THREAD

**Issue by iltish** [moby/moby#33829]
**Description**
Docker 17.06.0-ce-rc5 (also earlier 17.06 RC builds) breaks rancher-agent. It works without problems on all previous docker versions (from 1.10.3 to 17.05).

**Steps to reproduce the issue:**
1.  Start rancher-server on host1: `docker run -d --restart=unless-stopped -p 80:8080 rancher/server`
2.  Navigate to Infrastructure->Hosts->Add Host in rancher web-interface on host1 (`http://<host1>/`)
3.  Follow rancher instructions and launch rancher-agent on host2 with ` docker run --rm --privileged -v /var/run/docker.sock:/var/run/docker.sock -v /var/lib/rancher:/var/lib/rancher rancher/agent:v1.2.2 http://<host1>/v1/scripts/<key1>:<key2>:<key3>`

**Describe the results you received:**
rancher-agent keeps restarting with the following errors:
```
# docker ps
CONTAINER ID        IMAGE                  COMMAND             CREATED              STATUS                          PORTS               NAMES
14cabd80df8f        rancher/agent:v1.2.2   "/run.sh run"       About a minute ago   Restarting (1) 10 seconds ago                       rancher-agent
# docker logs rancher-agent
Found container ID: 011c0f599e468b2c4ef944f532ba1c3c5d20da4dbeb2a6203ffb522a42955d2e
Checking root: /host/run/runc
Checking file: 011c0f599e468b2c4ef944f532ba1c3c5d20da4dbeb2a6203ffb522a42955d2e
Found state.json: 011c0f599e468b2c4ef944f532ba1c3c5d20da4dbeb2a6203ffb522a42955d2e
time="2017-06-26T17:42:04Z" level=fatal msg="json: cannot unmarshal object into Go value of type []string" 
Found container ID: 151dddfb48cdd03f6fa65c948851b6203380e0c32da79733ccfb4f7807d64406
Checking root: /host/run/runc
Checking file: 14cabd80df8fc70773dbb12ca20e8d39d4bf61e1d28b08699106a0513562ae03
Checking file: 151dddfb48cdd03f6fa65c948851b6203380e0c32da79733ccfb4f7807d64406
Found state.json: 151dddfb48cdd03f6fa65c948851b6203380e0c32da79733ccfb4f7807d64406
time="2017-06-26T17:42:05Z" level=fatal msg="json: cannot unmarshal object into Go value of type []string"
...
```

**Describe the results you expected:**
rancher-agent starting all necessary rancher components and host2 being added to the list of available rancher hosts.

**Additional information you deem important (e.g. issue happens only occasionally):**
rancher-agent is currently working fine on all docker versions ranging from 1.10.3 to 17.05.0-ce, so this could to be related to break down of backward compatibility on docker side.

**Output of `docker version`:**
```
Client:
 Version:      17.06.0-ce-rc5
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   b7e4173
 Built:        Tue Jun 20 07:13:24 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.06.0-ce-rc5
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   b7e4173
 Built:        Tue Jun 20 07:12:15 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Output of `docker info`:**
```
Containers: 1
 Running: 0
 Paused: 0
 Stopped: 1
Images: 8
Server Version: 17.06.0-ce-rc5
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 29
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: cfb82a876ecc11b5ca0977d1733adbe58599088a
runc version: 2d41c047c83e09a6d61d464906feb2a2f3c52aa4
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.8.0-56-generic
Operating System: Ubuntu 16.04.2 LTS
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 3.613GiB
Name: rancher-client
ID: IZLZ:4FUS:7N4I:M6OP:WN7K:QR7R:AQI3:WVVW:JGO2:HSNA:5KKL:TK7Y
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
n1-standard-1 (1 vCPU, 3.75 GB memory) on GCE, but the same issues is reproducible with other environment configurations.

**Comment by thaJeztah** [comment]
ping @mlaventure @crosbymichael PTAL

**Comment by mlaventure** [comment]
@iltish the `runtime-spec` / `runc` combination was updated, and as such rancher would need to do so too if it parses/generates those.

`runtime-spec`: `v1.0.0-rc5`

**Comment by iltish** [comment]
Opened corresponding issue for rancher [rancher/rancher/issues/9192](https://github.com/rancher/rancher/issues/9192)

**Comment by ibuildthecloud** [comment]
This is being fixed in Rancher 1.6.3.  This is not a Docker issue, but a Rancher issue.  The Docker issue can be closed.

**Comment by thaJeztah** [comment]
Thanks for the update @ibuildthecloud! I wanted to check up with you on this, but got occupied with other things :heart:
